Security definer materialized view

marsou001 opened the pull request:
Fixes #40799 

Now, the correct error will be shown for materialized views accessible to `anon` and `authenticated` roles via Data API, instead of the unrelated "Security Definer view" error.

For convenience, users can immediately fix the issue by running the correct SQL query to revoke select from `anon` and `authenticated`, just by clicking on "confirm" on the confirmation modal I just created.


https://github.com/user-attachments/assets/f3ce9353-4ad0-4063-bf33-0b403f5fa87a

Before

<img width="958" height="440" alt="materialized" src="https://github.com/user-attachments/assets/89047c91-da35-4b9f-b7e3-82e877bcf2c6" />

Edit:

Thanks to the review by @saltcod, I now realise that revoking access from `anon` and `authenticated` users might not be the optimal solution since it would break many projects relying on the access to materialized views.

After pondering on the possible solutions, I figured there isn't an easy one, so I did away with the autofix button, and instead created a dialog explaining three possible options for the user, with a sample query under each one for convenience.

marsou001 committed:
fix(studio): show correct warning for materialized view exposed via API

marsou001 committed:
create dialog to auto fix  materialized view accessible via API

marsou001 committed:
use confirmation dialog to fix materialized view accessible via API warning

kallebysantos commented:
Hi @marsou001  💚
Thanks for your contributions, may you run `pnpm run format` to pass CI?

marsou001 committed:
run prettier

marsou001 commented:
@kallebysantos I ran the test, it has passed!

marsou001 committed:
remove autofix for materialized view lint
- Show dialog with two possible solutions for materialized view in API warning.

marsou001 committed:
prettier

marsou001 committed:
add third solution and queries below each solution
